airbyte-integrations/connectors/source-surveycto/source_surveycto/source.py

```diff
@@ -109,9 +109,26 @@ def read_records(self, *args, **kwargs) -> Iterable[Mapping[str, Any]]:
 
 # Source
 class SourceSurveycto(AbstractSource):
-    def check_connection(self, logger, config) -> Tuple[bool, any]:
-        return True, None
 
+    
+    def check_connection(self, logger, config) -> Tuple[bool, Any]:
+        
+        form_ids = config["form_id"]
+        
+        try:
+            for form_id in form_ids:
+                schema = Helpers.call_survey_cto(config, form_id)
+                filter_data = Helpers.get_filter_data(schema)
+                schema_res = Helpers.get_json_schema(filter_data)
+                stream = SurveyctoStream(config=config, form_id=form_id, schema=schema_res)
+                next(stream.read_records(sync_mode=SyncMode.full_refresh))
+                
+            return True, None
+        
+        except Exception as error:
+            return False, f"Unable to connect - {(error)}"
```

Comment:
@avirajsingh7 can you add unit test to validate the function. Airbyte doesn't have integration tests for this particulr connector and have a large unit test coverage can help keep the connector quality. After that we can merge the contribution:
* One returning True
* Other going to the exception

Thanks

Reply:
@marcosmarxm added unit testing